### PR TITLE
Eliza admin invoices

### LINKED ISSUE #44: User Story 32 - Admin Invoices

### LINKED ISSUE #45: User Story 33 - Admin Invoices

### LINKED ISSUE #46: User Story 34 - Admin Invoices

### LINKED ISSUE #47: User Story 35 - Admin Invoices

### LINKED ISSUE #48: User Story 36 - Admin Invoices

### PR DESCRIPTION
**Pull request by elizakeating**
## Description

Implemented the Admin Invoices (US 32-36) section.

US 32 includes:
- An admin invoices index page that shows the invoice ids of all invoices

US 33 includes:
- An admin invoice show page that shows:
   - invoice id
   - invoice status
   - invoice created_at date in format "Day, Month Date, Year"
   - customer of the invoice's first and last name

US 34 includes:
- A section on the invoice show page that showcases all of that invoice's items, including:
   - item name
   - quantity of item ordered
   - price the item sold for
   - invoice item status

US 35 includes:
- A line on the invoice show page that showcases that invoice's total revenue

US 36 includes:
- A select field for the invoice status and an 'Update Invoice Status' button that when clicked updates the invoice's status

## Type of change

- [ ] fix
- [x] feat
- [x] test
- [x] refactor
- [ ] docs

## Checklist

- [x] code has been self reviewed
- [x] code runs without any errors
- [x] thorough testing has been implemented if adding feature
- [x] all tests pass

### Thanks!

### CODE REVIEW

File: app/models/invoice_item.rb
Comment (DylanScotty) [inline app/models/invoice_item.rb:1]: I like how you did this and drew it out

mine was just invoice_items.sum('quantity*unit_price') not sure if they are doing the same thing

File: app/controllers/admin/invoices_controller.rb [hunk 1]
```diff
@@ -1,4 +1,17 @@
...
+  end
+
+  def show
+    @invoice = Invoice.find(params[:id])
+    @invoice_items = InvoiceItem.find_items(@invoice)
+  end
+
+  def update
+    invoice = Invoice.find(params[:id])
+    invoice.update!(status: params[:status])
```
Comment (lanebret85) [inline app/controllers/admin/invoices_controller.rb:12]: would we potentially want to consider putting :status into strong params in a private method at the bottom?

File: app/views/admin/invoices/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<h1>Invoice #<%= @invoice.id %></h1>
+
+<%= form_with url: "/admin/invoices/#{@invoice.id}", method: :patch, data: { turbo: false } do |form| %>
+  <%= form.label :status, "Status:" %>
+  <%= form.select :status, ["cancelled", "completed", "in progress"], selected: @invoice.status %>
```
Comment (lanebret85) [inline app/views/admin/invoices/show.html.erb:5]: I don't fully understand what's happening here, but I trust that you have it right! I also would really like to pick your brain about what's happening under the hood on this

**Review by lanebret85 — APPROVED** [review]
Great stuff!! I made a suggestion about moving the update of :status for admin invoices into strong params in the admin invoices controller, but all looks good to me

**Pull request merged**